File: Backend/MagicApp/Models/Dtos/WebSocketMessage.cs
using System.Text.Json.Serialization;

namespace MagicApp.Models.Dtos;

public enum MsgType
{
    Connection,             // Conexión con el WebSocket
    GlobalChat,             // Mensaje del chat global
    PrivateChat,            // Mensaje del chat privado
    UserBanned,             // Usuario baneado
    ForumNotification,      // Notificación del foro
    ChatNotification,       // Notificación de chat
    ChatWithAI,             // Mensaje del chat con la IA
    CardDetailsWithAI,      // Comentar una carta con la IA
    CancelAIMessage         // Cancelar petición hacia la IA
}

public class WebSocketMessage
{
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public MsgType Type { get; set; }

    public object Content { get; set; }
}

File: Frontend/MagicApp/src/app/models/web-socket-message.ts
export enum MsgType {
    Connection = 'Connection',
    GlobalChat = 'GlobalChat',
    PrivateChat = 'PrivateChat',
    UserBanned = 'UserBanned',
    ForumNotification = 'ForumNotification',
    ChatNotification = 'ChatNotification',
    ChatWithAI = 'ChatWithAI',
    CardDetailsWithAI = 'CardDetailsWithAI',
    CancelAIMessage = 'CancelAIMessage'
}

export interface WebSocketMessage {
    Type: MsgType;
    Content: any;
}